### PR TITLE
Enable SpatialD tests on MacOS

### PR DESCRIPTION
**Pull request by zeroZshadow**
Magically the tests have started working consistently.

It seems the machine needed the reboot that happened by moving the machine from EV's desk to the server rack?

### CODE REVIEW

File: .buildkite/premerge.steps.yaml [hunk 1]
```diff
@@ -73,16 +73,28 @@ steps:
...
+        allow_failure: true
     plugins:
       - improbable/test-summary#d7289cac8297018fd1b452dcf828979307b3ebc6:
           inputs:
             - label: ":octagonal_sign: :windows: Test failures "
               artifact_path: "logs\\nunit\\*.xml"
               type: nunit
+            - label: ":octagonal_sign: :darwin: Test failures "
+              artifact_path: "logs/nunit/*.xml"
+              type: nunit
```
Comment (jamiebrynes7) [inline .buildkite/premerge.steps.yaml]: Its funny that this oddity in BuildKite actually helps us out here

File: ci/test.sh [hunk 2]
```diff
@@ -20,6 +20,7 @@ dotnet test \
     --logger:"nunit;LogFilePath=${TEST_RESULTS_DIR}/code-gen-lib-test-results.xml" \
     workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeGenerationLib.csproj
 
+echo ""
```
Comment (jamiebrynes7) [inline ci/test.sh]: Add a comment why this exists

File: .buildkite/premerge.steps.yaml (hunk 1 above)
Comment (zeroZshadow) [reply .buildkite/premerge.steps.yaml]: I'm not 100% sure this actually works as intended.

Comment (jamiebrynes7) [reply .buildkite/premerge.steps.yaml]: Force a test failure (in different ways for each platform) and see what happens?

Comment (zeroZshadow) [reply .buildkite/premerge.steps.yaml]: Just going to enable the test first. will test do another pass for the annotation afterwards

File: ci/test.sh (hunk 2 above)
Comment (zeroZshadow) [reply ci/test.sh]: Done

**Pull request merged**